src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  constructor(name, health) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+  bite(animal) {
+    if (animal.hasOwnProperty('hidden')) {
+      if (!animal.hidden) {
+        animal.health -= 50;
+      }
+
+      if (animal.health <= 0) {
+        Animal.alive = Animal.alive.filter(
+          (aliveAnimal) => aliveAnimal !== animal,
+        );
+      }
+    }
```

Comment:
[CODE QUALITY] - The `bite` method implementation could be simplified by combining the checks for whether the animal is hidden and whether it is a Herbivore into a single condition. This would improve readability and efficiency of the code. For example, `if (animal instanceof Herbivore && !animal.hidden) { ... }`.